### ISSUE TITLE
Unable to upgrade from 2.4.4

### ISSUE THREAD

**Issue by FlorentCoppint** [alandtse/tesla#390]
Hi,

I use Tesla Custom Integration in my Home Assistant.
In version 2.4.4, it works very well.

When I try to upgrade to a newer version, integration does not work anymore (no entity available).
I don't have any error message, or don't know where to find them.

If I downgrade to 2.4.4, it works again without touching anything.

I don't want to remove/reset integration, I don't want to lose history.

Where can I find debug information about my issue ?

Thank you.

**Comment by alandtse** [comment]
It's a breaking change. Hence the version number change. You either update and redo the entities or you stay with the old one which is unsupported.

**Comment by FlorentCoppint** [comment]
Oh ok... :( so bad.
Is it possible to export & re-import history of entities in that case ? :(

**Comment by carleeno** [comment]
With a bit of work it's possible to keep history for some/most of the entities.
TL;DR: If you rename the new entities to match the old ones, they should retain the history of the old ones.

Before you upgrade, go to Settings -> Devices and Services -> Entities and filter by the name of your car to see all the old entities, make a note of all of their entity IDs

After you upgrade, do the same, but click each entity and click Settings to change the entity ID to match the old name. This will only work on entities that have the same domain as before (eg you can't change lock.frunk to cover.frunk). I know some sensors were added, some were removed, some were changed; so it won't be perfect.

I should note this is an uncommon practice to rename the entity ids, if you reach out for help and mention a renamed entity id, we might get confused :)
